integration-tests/smoke/ccip/fee_boosting_test.go

```diff
@@ -1,121 +1,303 @@
 package smoke
 
 import (
+	"context"
+	"fmt"
 	"math/big"
 	"testing"
 	"time"
 
+	"cosmossdk.io/errors"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/test-go/testify/require"
 	"golang.org/x/exp/maps"
 
+	"github.com/smartcontractkit/chainlink-common/pkg/utils/tests"
+	"github.com/smartcontractkit/chainlink-testing-framework/lib/utils/testcontext"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/ccip/ccipevm"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/fee_quoter"
+
+	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
 	"github.com/smartcontractkit/chainlink/deployment"
 	"github.com/smartcontractkit/chainlink/deployment/ccip/changeset"
 	testsetups "github.com/smartcontractkit/chainlink/integration-tests/testsetups/ccip"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/onramp"
 	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/router"
 	"github.com/smartcontractkit/chainlink/v2/core/logger"
 )
 
-type feeboostTestCase struct {
-	t                      *testing.T
-	sender                 []byte
-	deployedEnv            changeset.DeployedEnv
-	onchainState           changeset.CCIPOnChainState
-	initialPrices          changeset.InitialPrices
-	priceFeedPrices        priceFeedPrices
-	sourceChain, destChain uint64
-}
-
-type priceFeedPrices struct {
-	linkPrice *big.Int
-	wethPrice *big.Int
-}
+var (
+	linkPrice = deployment.E18Mult(100)
+	wethPrice = deployment.E18Mult(4000)
+)
 
-// TODO: find a way to reuse the same test setup for all tests
 func Test_CCIPFeeBoosting(t *testing.T) {
-	setupTestEnv := func(t *testing.T, numChains int) (changeset.DeployedEnv, changeset.CCIPOnChainState, []uint64) {
-		e, _, _ := testsetups.NewLocalDevEnvironment(t, logger.TestLogger(t), deployment.E18Mult(5), big.NewInt(9e8), nil)
+	e, _, _ := testsetups.NewLocalDevEnvironment(t, logger.TestLogger(t), linkPrice, wethPrice, &changeset.TestConfigs{
+		IsFeeBoosting: true,
+	})
```

Comment:
Can we see if this test can be run using the sim chain? It would greatly increase the speed of the test

Reply:
250s vs 70s.